Rollbar/RollbarNotifier.m

```diff
@@ -331,26 +341,32 @@ - (BOOL)sendPayload:(NSData*)payload {
     [request setValue:self.configuration.accessToken forHTTPHeaderField:@"X-Rollbar-Access-Token"];
     [request setHTTPBody:payload];
     
-    NSError *error;
-    NSHTTPURLResponse *response;
+//    NSError *error;
+//    NSHTTPURLResponse *response;
+//    NSData *data = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:&error];
     
-    NSData *data = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:&error];
+    dispatch_semaphore_t sem = dispatch_semaphore_create(0);
     
-    if (error) {
-        RollbarLog(@"There was an error reporting to Rollbar");
-        RollbarLog(@"Error: %@", [error localizedDescription]);
-    } else {
-        NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
-        if ([httpResponse statusCode] == 200) {
-            RollbarLog(@"Success");
-            return YES;
+    __block BOOL result = NO;
+    NSURLSessionDataTask *dataTask = [[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (error) {
+            RollbarLog(@"There was an error reporting to Rollbar");
+            RollbarLog(@"Error: %@", [error localizedDescription]);
         } else {
-            RollbarLog(@"There was a problem reporting to Rollbar");
-            RollbarLog(@"Response: %@", [NSJSONSerialization JSONObjectWithData:data options:0 error:nil]);
+            NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+            if ([httpResponse statusCode] == 200) {
+                RollbarLog(@"Success");
+                result = YES;
+            } else {
+                RollbarLog(@"There was a problem reporting to Rollbar");
+                RollbarLog(@"Response: %@", [NSJSONSerialization JSONObjectWithData:data options:0 error:nil]);
+            }
         }
-    }
+    }];
+    [dataTask resume];
+    dispatch_semaphore_wait(sem, dispatch_time(DISPATCH_TIME_NOW, 5*NSEC_PER_SEC));
```

Comment:
Where is the `signal` this is waiting for? This looks like a glorified sleep(5) at the moment.

Reply:
Oops, added `dispatch_semaphore_signal`.